.travis.yml

```
@@ -52,18 +52,33 @@ matrix:
       env:
         - DEPS=latest
 
+before_install:
+  - shopt -s expand_aliases
+  - alias composer="travis_retry docker-compose run --rm php composer"
+  - alias php="travis_retry docker-compose run -e TRAVIS -e TRAVIS_JOB_ID --rm php php"
+  - docker-compose build --build-arg PHP_VERSION=${TRAVIS_PHP_VERSION} --build-arg XDEBUG=${TEST_COVERAGE:+1} --no-cache php
+  - composer --version
+  - php -v
+  - php -m
+
 install:
+  - php --info
   - composer install $COMPOSER_ARGS --ignore-platform-reqs
+  - if [[ $LEGACY_DEPS != '' ]]; then composer update $COMPOSER_ARGS --with-dependencies $LEGACY_DEPS ; fi
   - if [[ $DEPS == 'latest' ]]; then travis_retry composer config "platform.ext-mongo" "1.6.16" && composer update $COMPOSER_ARGS ; fi
   - if [[ $DEPS == 'lowest' ]]; then travis_retry composer config "platform.ext-mongo" "1.6.16" && composer update --prefer-lowest --prefer-stable $COMPOSER_ARGS ; fi
-
-  - travis_retry composer config "platform.ext-mongo" "1.6.16" && composer require --dev $COMPOSER_ARGS $ADAPTER_DEPS
+  - if [[ $TEST_COVERAGE == 'true' ]]; then composer require --dev $COMPOSER_ARGS $COVERAGE_DEPS ; fi
+  - travis_retry composer config "platform.ext-mongo" "1.6.16" && composer require --dev $COMPOSER_ARGS 
   - travis_retry composer config "platform.ext-mongo" "1.6.16" && composer install $COMPOSER_ARGS
   - stty cols 120 && composer show
+  - composer show
 
 script:
   - cp ./tests/TestConfiguration.php ./config/application.config.php
   - ./vendor/bin/doctrine-module odm:schema:create
   - ./vendor/bin/doctrine-module odm:schema:drop
-  - if [[ $TEST_COVERAGE == 'true' ]]; then ./vendor/bin/phpunit --coverage-clover clover.xml ; else ./vendor/bin/phpunit ; fi
-  - if [[ $CHECK_CS == 'true' ]]; then ./vendor/bin/phpcs --standard=PSR2 ./src/ ./tests/ ./config/ ; fi
+  - if [[ $TEST_COVERAGE == 'true' ]]; then php vendor/bin/phpunit --coverage-clover clover.xml ; else php vendor/bin/phpunit ; fi
+  - if [[ $CHECK_CS == 'true' ]]; then php vendor/bin/phpcs --standard=PSR2 ./src/ ./tests/ ./config/ ; fi
```

```suggestion
  - if ${CHECK_CS=false}; then php vendor/bin/phpcs --standard=PSR2 ./src/ ./tests/ ./config/ ; fi
```

or even

```suggestion
  - if $CHECK_CS; then php vendor/bin/phpcs --standard=PSR2 ./src/ ./tests/ ./config/ ; fi
```

should work, since `true` is a binary that exits with a successful exit code (and `false` is the opposite)